graftm/hmmer.py

```diff
@@ -992,18 +994,19 @@ def search_and_extract_nucleotides_matching_nucleotide_database(self,
                         hits[h[0]]={"strand":[h[1]],
                                     "span":[[h[2], h[3]]],
                                     "query_span":[[h[4], h[5]]]}
+
             hit_readnames = hits.keys()
             
             if euk_check:
                 euk_reads = self._check_euk_contamination(table_list)
-                euk_reads = set(euk_reads)
-                prok_reads = set([hit_readnames for read in hit_readnames\
-                                  if read not in euk_reads])
-                for read in prok_reads: readnames.write(read + '\n')
-                hit_read_count = [len(euk_reads), len(prok_reads)]
-            else:
-                [readnames.write(read + '\n') for read in hit_readnames]
-                hit_read_count = [0, len(hit_readnames)]
+                hit_readnames = set([read for read in hit_readnames if read not in euk_reads])
+                for read in hit_readnames: 
+                    readnames.write(read + '\n')
+                hits = {key:item for key, item in  hits.iteritems() if key in hit_readnames}
+                hit_read_count = [len(euk_reads), len(hit_readnames)]
+            for read in hit_readnames: 
+                readnames.write(read + '\n')
+            hit_read_count = [0, len(hit_readnames)]
             readnames.flush()
             
             hit_reads_fasta, complement_information = self._extract_from_raw_reads(
```

Comment:
probably would be better to not the list of reads to extract to file, but rather do it like we do elsewhere, but meh